docker/istio-dev/Dockerfile

```diff
@@ -0,0 +1,73 @@
+FROM docker:stable as docker
+
+FROM ubuntu:xenial
+ARG goversion=1.12.5
```

Comment:
kubectl is tricky since only one version of `kubectl` is ever available from the APT repo at a time. And only a few versions of Cloud SDK are available from the repo.

Added KIND version, and updated the Go version.